### ISSUE TITLE
Refactoring and logging related changes

### ISSUE THREAD

**Issue by vaibhav-yb** [yugabyte/debezium-connector-yugabytedb#111]
This open issue will keep track of the refactoring or any other minor changes required.

**Comment by vaibhav-yb** [comment]
Investigate why the connector prints partition details after `Finished streaming` log:
```
2022-12-08 06:15:25,755 INFO   YugabyteDB|test_server|streaming  Finished streaming   [io.debezium.pipeline.ChangeEventSourceCoordinator]
2022-12-08 06:15:25,755 INFO   YugabyteDB|test_server|streaming  YBPartition is YBPartition{tabletId='9d8d21450edf4454aad0b3da4e42389c'} and YugabyteDBOffsetContext while streaming is null   [io.debezium.connector.yugabytedb.YugabyteDBChangeEventSourceCoordinator]
```

**Comment by vaibhav-yb** [comment]
Add task ID in the logging context.

**Comment by vaibhav-yb** [comment]
Currently, we handle tablet split in the order:
1. Remove parent tablet from polling list
2. Get child tablets and add them to the list
In case step 2 failed and we went for a retry, we will not be left with the parent tablet in the list since it has been removed. 

Now what we need to do is:
1. Get child tablets for the parent
2. Remove the parent tablet from polling list

This will ensure that if the API to fetch child tablet fails for some reasons, we are not left with no option to get the parent further.
